### DISCUSSION TITLE
Is there a option to lazy load the geomtry from an ifc file

### DISCUSSION THREAD

**Discussion by Martin15135215** [IfcOpenShell/IfcOpenShell#2804] [General]
At the moment you do this `model = ifcopenshell.open("test.ifc")`, all of the ifc file is loaded, however most of the time I am only interested in the psets and the elements, not the representation (geometry of an element). Sadly, I have a.ifc file that is huge and bloated because of geometry (+200MB). Because of that reading and writing ifc files are also very slow, even if I just want to change the IfcSite Name in a for loop.

Are there ways to avoid loading or lazy loading the geometry of an ifc file? Like `model = ifcopenshell.open("test.ifc" , lazy_load_representation=True)` → that would immensely speed up my little scripts.

**Comment by Moult** [comment]
No :(

https://github.com/IfcOpenShell/IfcOpenShell/issues/2025

Or deal with it as text?

**Comment by aothms** [comment]
Note that the parser is to some extent lazy, but not as lazy as it used to be (for building the guid map for example). I think the parser will give you something like 10 - 20 mb/s which I don't think is bad. Dealing with it as text (sed / grep) will do better, I don't know maybe up to 100mb / s (I'm making up numbers here, would be cool to have these numbers available and compared to other parsers). I think dealing with large text blobs is slow regardless. There simply isn't a lot of magic possible on unstructured text.
